### ISSUE TITLE
MySQL reflection CONSTRAINT FOREIGN KEY ON DELETE is missing SET DEFAULT

### ISSUE THREAD

**Issue by quentin-roche** [sqlalchemy/sqlalchemy#11285]
### Describe the bug

Hello,
This is a very small issue but quite annoying when creation automatic migration in alembic.
 
`CONSTRAINT FOREIGN KEY` `ON DELETE` `SET DEFAULT` is missing from mysql reflection.
MySQL [documentation](https://dev.mysql.com/doc/refman/8.3/en/create-table-foreign-keys.html) for FOREIGN KEY Constraints.

The issue is [here](https://github.com/sqlalchemy/sqlalchemy/blob/b5cf61c504e6ff7cdceeb0ca376eb47a97b9da5a/lib/sqlalchemy/dialects/mysql/reflection.py#L501-L521):

``` python
        # CONSTRAINT `name` FOREIGN KEY (`local_col`)
        # REFERENCES `remote` (`remote_col`)
        # MATCH FULL | MATCH PARTIAL | MATCH SIMPLE
        # ON DELETE CASCADE ON UPDATE RESTRICT
        #
        # unique constraints come back as KEYs
        kw = quotes.copy()
+       kw["on"] = "RESTRICT|CASCADE|SET NULL|NO ACTION"
-       kw["on"] = "RESTRICT|CASCADE|SET NULL|NO ACTION|SET DEFAULT"
        self._re_fk_constraint = _re_compile(
            r"  "
            r"CONSTRAINT +"
            r"%(iq)s(?P<name>(?:%(esc_fq)s|[^%(fq)s])+)%(fq)s +"
            r"FOREIGN KEY +"
            r"\((?P<local>[^\)]+?)\) REFERENCES +"
            r"(?P<table>%(iq)s[^%(fq)s]+%(fq)s"
            r"(?:\.%(iq)s[^%(fq)s]+%(fq)s)?) +"
            r"\((?P<foreign>(?:%(iq)s[^%(fq)s]+%(fq)s(?: *, *)?)+)\)"
            r"(?: +(?P<match>MATCH \w+))?"
            r"(?: +ON DELETE (?P<ondelete>%(on)s))?"
            r"(?: +ON UPDATE (?P<onupdate>%(on)s))?" % kw
        )
```

### SQLAlchemy Version in Use

2.0.7

### DBAPI (i.e. the database driver)

mysqlclient

### Database Vendor and Major Version

MySQL 8

### Python Version

3.11

### Operating system

MacOS

### To Reproduce

```python
python
from sqlalchemy import create_engine, text, MetaData

DATABASE_CONNECTION_STRING = "mysql://root:password@127.0.0.1/reflection_issue"

engine = create_engine(DATABASE_CONNECTION_STRING, echo=True)

# Create tables
with engine.connect() as conn:
    conn.execute(text("""
        CREATE TABLE foreign_table (
            id INT PRIMARY KEY AUTO_INCREMENT
        )
    """))
    conn.execute(text("""
        CREATE TABLE test_table (
            id INT PRIMARY KEY AUTO_INCREMENT,
            foreign_id_set_default INT,
            foreign_id_set_null INT,
            FOREIGN KEY (foreign_id_set_null) REFERENCES foreign_table(id) ON DELETE SET NULL,
            FOREIGN KEY (foreign_id_set_default) REFERENCES foreign_table(id) ON DELETE SET DEFAULT
        )
    """))


# Reflect tables
with engine.connect() as conn:
    metadata = MetaData()
    metadata.reflect(bind=conn)

    set_null = list(metadata.tables["test_table"].columns["foreign_id_set_null"].foreign_keys)[0].ondelete
    assert set_null == "SET NULL", f"Expected SET NULL, got {set_null}"
    set_default = list(metadata.tables["test_table"].columns["foreign_id_set_default"].foreign_keys)[0].ondelete
    assert set_default == "SET DEFAULT", f"Expected SET DEFAULT, got {set_default}"
```

### Error

```
Traceback (most recent call last):
  File "/Users/rochequentin/PycharmProjects/ptx-project-optimizer/reflection_issue.py", line 33, in <module>
    assert set_default == "SET DEFAULT", f"Expected SET DEFAULT, got {set_default}"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Expected SET DEFAULT, got None
```

**Comment by CaselIT** [comment]
Hi thanks for reporting. Care to open a PR to fix?

**Comment by zzzeek** [comment]
didnt we have this issue reported recently already?

**Comment by zzzeek** [comment]
oh well, guess not

**Comment by CaselIT** [comment]
no that was a mysql dialect that did things differently

**Comment by CaselIT** [comment]
@zzzeek you may be thinking of https://github.com/sqlalchemy/sqlalchemy/discussions/11194

**Comment by zzzeek** [comment]
hm.   doesnt feel familiar but maybe

**Comment by quentin-roche** [comment]
> Hi thanks for reporting. Care to open a PR to fix?

Sure I will do that next week

**Comment by sqla-tester** [comment]
**roche-quentin** has proposed a fix for this issue in the **main** branch:

**FIX #11285 CONSTRAINT FOREIGN KEY ON DELETE was missing SET DEFAULT from mysql reflection** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5297

**Comment by sqla-tester** [comment]
**roche-quentin** has proposed a fix for this issue in the **rel_2_0** branch:

**Add ``SET DEFAULT`` reflection option** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5298
